packages/pds/src/sql-repo-storage.ts

```diff
@@ -191,56 +191,65 @@ export class SqlRepoStorage extends ReadableBlockstore implements RepoStorage {
           indexedAt: this.getTimestamp(),
         })
         .onConflict((oc) =>
-          oc.column('did').doUpdateSet({ root: cid.toString() }),
+          oc.column('did').doUpdateSet({
+            root: cid.toString(),
+            indexedAt: this.getTimestamp(),
+          }),
         )
         .execute()
     }
   }
 
-  async getCarStream(rev?: string) {
+  async getCarStream(since?: string) {
     const root = await this.getRoot()
     if (!root) {
       throw new RepoRootNotFoundError()
     }
     return writeCarStream(root, async (car) => {
-      let blocks = await this.getBlockRange(500, rev)
+      let cursor: RevCursor | undefined = undefined
       do {
-        for (const block of blocks) {
-          await car.put(block)
+        const res = await this.getBlockRange(since, cursor)
+        for (const row of res) {
+          await car.put({
+            cid: CID.parse(row.cid),
+            bytes: row.content,
+          })
         }
-        const cidCursor = blocks.at(-1)?.cid
-        if (!cidCursor) {
-          break
+        const lastRow = res.at(-1)
+        if (lastRow && lastRow.repoRev) {
+          cursor = {
+            cid: CID.parse(lastRow.cid),
+            rev: lastRow.repoRev,
+          }
+        } else {
+          cursor = undefined
         }
-        blocks = await this.getBlockRange(500, rev, cidCursor)
-      } while (blocks.length > 0)
+      } while (cursor)
     })
   }
 
-  async getBlockRange(
-    limit: number,
-    rev?: string,
-    cidCursor?: CID,
-  ): Promise<{ cid: CID; bytes: Uint8Array }[]> {
+  async getBlockRange(since?: string, cursor?: RevCursor) {
     let builder = this.db.db
       .selectFrom('ipld_block')
       .where('creator', '=', this.did)
+      .select(['cid', 'repoRev', 'content'])
       .orderBy('repoRev', 'asc')
       .orderBy('cid', 'asc')
-      .select(['cid', 'content'])
-      .limit(limit)
-
-    if (rev) {
-      builder = builder.where('repoRev', '>=', rev)
-    }
-    if (cidCursor) {
-      builder = builder.where('cid', '>', cidCursor.toString())
+      .limit(500)
+    if (cursor) {
+      builder = builder.where((qb) =>
+        qb
+          .where('repoRev', '>', cursor.rev)
+          .orWhere((inner) =>
+            inner
+              .where('repoRev', '=', cursor.rev)
+              .where('cid', '>', cursor.cid.toString()),
+          ),
```

Comment:
Would recommend using this pattern to ensure you hit the index: `("repoRev", "cid") > (${cursor.rev}, ${cursor.cid})`.  Shouldn't really be necessary, but we've seen it miss the index the other way for sure 🙃

Reply:
ah good looks 👍  simpler too